### PR TITLE
feat(lua): much more complete imgui bindings

### PR DESCRIPTION
**Pull request by xiaoxiao921**
allow lua imgui callbacks from outside yimmenu classic tabs through `gui.add_imgui(func)`
allow lua imgui callbacks inside yimmenu classic tabs through `tab:add_imgui(func)`

### CODE REVIEW

**Comment by gir489returns** [comment]
Build is failing because heap space is exhausted, doesn't seem like something the commit broke, seems like something GitHub needs to address. Build completed fine on my machine.

File: docs/lua/tables/gui.md [hunk 1]
```diff
@@ -79,4 +79,30 @@ gui.show_error(title, message)
...
+```
+
+- **Parameters:**
+  - `imgui_rendering` (function): Function that will be called every rendering frame, you can call ImGui functions in it, please check the ImGui.md documentation file for more info.
+
+**Example Usage:**
+```lua
+gui.add_imgui(imgui_rendering)
+```
+
```
Comment (Yimura) [inline docs/lua/tables/gui.md:107]: Also exists under `classes/tab.md`, is this intented?

**Comment by Yimura** [comment]
Doomed to CI failure with all the Lua bindings...

File: docs/lua/tables/gui.md (hunk 1 above)
Comment (xiaoxiao921) [reply docs/lua/tables/gui.md:107]: gui.

tab:

Not the same,though the descriptions could be clearer

Comment (Yimura) [reply docs/lua/tables/gui.md:107]: Oh yeah I see, in that way.
One draws to a Tab and the other is meant for raw drawing a UI with ImGui::Begin().

**Comment by gir489returns** [comment]
lmao fucking what? It builds now?

**Review by Yimura — APPROVED** [review]

**Pull request merged**